TypeCobol/Compiler/CupCommon/CobolWordsTypes.cs

```diff
@@ -21,6 +22,37 @@ public class PairTokenList : List<Tuple<Token,Token>> { };
     /// </summary>
     public class PairTokenListList : List<Tuple<List<Token>, List<Token>>> { };
 
+    /// <summary>
+    /// 
+    /// </summary>
+    public class CupReplaceOperation
+    {
+        public List<Token> From { get; }
+        public List<Token> By { get; }
+
+        public Token Leading {get; }
+        public Token Trailing { get; }
+
+        public CupReplaceOperation(List<Token> from, List<Token> by, Token leading, Token trailing)
+        {
+            From = from;
+            By = by;
+            Leading = leading;
+            Trailing = trailing;
+        }
+    }
+
+    public class CupReplaceOperations : List<CupReplaceOperation>
```

Comment:
Role ? This class does not add anything to a list of `CupReplaceOperation`

Reply:
Yes, but it seems that the cup grammar cannot handle directly a List<CupReplaceOperation>